src/blockdev.rs

```diff
@@ -809,6 +809,50 @@ pub fn lsblk_single(dev: &Path) -> Result<HashMap<String, String>> {
     Ok(devinfos.remove(0))
 }
 
+fn lsblk_all_and_reread_partition_tables() -> Result<Vec<HashMap<String, String>>> {
+    let mut cmd = Command::new("lsblk");
+    cmd.arg("--noheadings")
+        .arg("--nodeps")
+        .arg("--list")
+        .arg("--paths")
+        .arg("--output")
+        .arg("NAME");
+    let output = cmd_output(&mut cmd)?;
+    for dev in output.lines() {
+        if let Ok(mut fd) = std::fs::File::open(dev) {
+            let _ = reread_partition_table(&mut fd);
+        }
+    }
+    udev_settle()?;
+    let mut result: Vec<HashMap<String, String>> = Vec::new();
+    for dev in output.lines() {
+        let mut info = lsblk(Path::new(dev), true)?;
+        result.append(&mut info);
+    }
+    Ok(result)
+}
+
+pub fn get_filesystems_with_label(label: &str) -> Result<Vec<String>> {
+    let mut uuids = HashSet::new();
+    let result = lsblk_all_and_reread_partition_tables()?
+        .iter()
+        .filter(|v| v.get("LABEL").map(|l| l.as_str()) == Some(label))
+        .filter(|v| match v.get("UUID") {
+            Some(uuid) => {
+                if !uuid.is_empty() {
+                    uuids.insert(uuid)
```

Comment:
This function should probably have a docstring noting that if multiple filesystems with the label have the same UUID, we only return one of them.